bcgov/GDX-Analytics
# GDXDSD-7307-react-and-vue-js-tracker-dependabot-fixes

This PR updates the react_tracker/package-lock.json and vue_js_tracker/package-lock.json with the new versions of the following packages:

(react)
- "micromatch" to "4.0.8"
- "path-to-regexp": "0.1.10"
- "body-parser": "1.20.3"
- "serve-static": "1.16.2"

(vue)

- "send": "0.19.0"

The update should solve the dependabot alerts [#164](https://github.com/bcgov/GDX-Analytics/security/dependabot/164) [#165](https://github.com/bcgov/GDX-Analytics/security/dependabot/165) [#166](https://github.com/bcgov/GDX-Analytics/security/dependabot/166) [#167](https://github.com/bcgov/GDX-Analytics/security/dependabot/167) [#168](https://github.com/bcgov/GDX-Analytics/security/dependabot/168) [#169](https://github.com/bcgov/GDX-Analytics/security/dependabot/169) and [#170](https://github.com/bcgov/GDX-Analytics/security/dependabot/170).

Testing instructions:
(vue.js)
- clone the repo and switch to branch
```
git clone https://github.com/bcgov/GDX-Analytics.git
cd GDX-Analytics/
git checkout GDXDSD-7307-react-vue-tracker-dependabot-fixes
```
- install global packages for vue (omit warnings)
```
npm install -g @vue/cli
npm i -g vue-template-compiler
```
- install the dependencies listed in the package-lock.json for vue.js
```
cd examples/vue_js_tracker/ (omit warnings)
npm ci
```
- start a development server
`npm run serve`
- check if sample vue.js page is running by browsing to http://localhost:8080/

(react.js)
- stop server (ctrl +c)
- go to react folder
`cd ../react_tracker/`
- install global packages for react
`npm install -g react react-dom react-router-dom`
- install the dependencies listed in the package-lock.json for react.js (omit warnings)
`npm ci `
- start server
`npm start`
- check if sample react.js page is running by browsing to http://localhost:3000/
